prov/gni/src/gnix_msg.c

```diff
@@ -901,27 +1074,33 @@ static int __gnix_rndzv_iov_req_build(void *arg)
 	struct gnix_fid_ep *ep = req->gnix_ep;
 	struct gnix_nic *nic = ep->nic;
 	gni_ep_handle_t gni_ep = req->vc->gni_ep;
-	struct gnix_tx_descriptor *txd = NULL;
-	size_t recv_len, get_len, ct_size, send_cnt, recv_cnt, txd_cnt;
-	uint64_t recv_ptr = 0UL;
+	struct gnix_tx_descriptor *txd = NULL, *ct_txd = NULL;
+	size_t recv_len, send_len, get_len, send_cnt, recv_cnt, txd_cnt, ht_len;
+	uint64_t recv_ptr = 0UL, send_ptr = 0UL;
 	/* TODO: Should this be the sender's rndzv thresh instead? */
 	size_t rndzv_thresh = ep->domain->params.msg_rendezvous_thresh;
 	gni_ct_get_post_descriptor_t *cur_ct = NULL;
 	void **next_ct = NULL;
+	int head_off, head_len, tail_len;
 
 	GNIX_TRACE(FI_LOG_EP_DATA, "\n");
 	/*
 	 * TODO: xpmem intercept here
 	 */
 
-	if (req->vc->modes & GNIX_VC_MODE_XPMEM)
-		return  __gnix_rndzv_req_xpmem(req);
+	/* if (req->vc->modes & GNIX_VC_MODE_XPMEM) */
+	/* 	return  __gnix_rndzv_req_xpmem(req); */
```

Comment:
> Is this supposed to be commented out?

It was but after testing #905 with these changes, I will uncomment as the hangs have been resolved.